### PR TITLE
Sharable model and cancel button

### PR DESCRIPTION
**Pull request by fionaphillips**
Resolves [CG-42](https://unosquarebelfast.atlassian.net/secure/RapidBoard.jspa?rapidView=12&projectKey=CG&modal=detail&selectedIssue=CG-42)

basicModal

### CODE REVIEW

File: src/components/common/Modal/container.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      const { isOpen, title, bodyText, children, className } = this.props;
+
+      return (
+        <Wrapped isOpen={isOpen} title={title} bodyText={bodyText} className={className}>
+          {children}
+        </Wrapped>
+      );
+    }
+  };
+/* eslint-enable react/prefer-stateless-function */
```
Comment (girwin90) [inline src/components/common/Modal/container.js]: We should change these in the eslint js file.

File: src/components/common/Modal/modal.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import Modal from 'react-modal';
+import { PropTypes as PT } from 'prop-types';
+import container from './container';
+
+if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#root');
```
Comment (girwin90) [inline src/components/common/Modal/modal.js:6]: How come we need to do this?

Comment (fionaphillips) [reply src/components/common/Modal/modal.js:6]: @girwin90 this is a fix for a react-modal error that is being thrown in the tests https://github.com/reactjs/react-modal/issues/632

File: src/components/common/Modal/styled.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import styled from 'styled-components';
+import { device } from 'Themes';
+import ReactModalAdapter from './modal';
+
+export const StyledModal = styled(ReactModalAdapter)`
+  &__overlay {
```
Comment (girwin90) [inline src/components/common/Modal/styled.js:6]: Mentioned it in a previous PR, ideally you wouldn't be using classNames for these. Styled components is kinda trying to follow the React components. Small components that you break up and reuse.

File: src/pages/Selector/index.jsx [hunk 4]
```diff
@@ -37,6 +37,10 @@ const Selector = ({ employees, templates }) => (
       />
       <GenerateCVButton />
     </div>
+    <Modal isOpen title="Preview Document" bodyText={mockDataForModal.bodyText}>
+      <CancelModalButton className="cancel-button" />
+      <DownloadModalButton className="download-button" />
```
Comment (girwin90) [inline src/pages/Selector/index.jsx]: As mentioned above about styled components. Is the DownloadModalButton a styled component?

**Review by girwin90 — CHANGES_REQUESTED** [review]
A few questions.

File: src/components/common/Modal/styled.js (hunk 3 above)
Comment (fionaphillips) [reply src/components/common/Modal/styled.js:6]: @girwin90 These classNames were not created but came with the react-modal library that I am using. I need to amend the styling and position of the modal which is why I've referenced these classes here.

File: src/pages/Selector/index.jsx (hunk 4 above)
Comment (fionaphillips) [reply src/pages/Selector/index.jsx]: I've removed these

**Review by BrownMark — APPROVED** [review]
LGTM :shipit:

**Pull request merged**